server/controllers/finance/referenceGroup.js

```diff
@@ -0,0 +1,115 @@
+var db = require('../../lib/db');
+
+function lookupReferenceGroup(id, codes) {
+  'use strict';
+
+  var sql =
+    'SELECT id, reference_group, text, position, section_bilan_id FROM reference_group WHERE id = ?';
+
+  return db.exec(sql, id)
+    .then(function (rows) {
+      if (rows.length === 0) {
+        throw new codes.ERR_NOT_FOUND();
```

Comment:
This method of throwing errors is difficult to understand when reading code. (Where did `codes` come from and how is it different from `id`?). 

This is perfectly valid for how we used to handle server errors however there was been an important update
https://github.com/IMA-WorldHealth/bhima-2.X/issues/182 was resolved with https://github.com/IMA-WorldHealth/bhima-2.X/pull/208

We can now use: 
```js
var NotFound = require('lib/errors/NotFound');
...
throw new NotFound('The entity was not found');
```

This is much clearer to read and gives additional benefits like stack traces.